livre/lib/render/AvailableSetGenerator.cpp

```diff
@@ -29,155 +29,156 @@
 #include <livre/lib/visitor/CollectionTraversal.h>
 #include <livre/lib/visitor/DFSTraversal.h>
 #include <livre/lib/cache/TextureObject.h>
+#include <livre/lib/cache/TextureCache.h>
 
 namespace livre
 {
 
-typedef boost::unordered_map< NodeId, dash::NodePtr > NodeIdDashNodeMap;
-
 class VisibleCollectorVisitor : public RenderNodeVisitor
 {
 public:
 
     VisibleCollectorVisitor( DashTreePtr dashTree,
-                             DashNodeVector& nodeVector );
-
-    void visit( DashRenderNode& renderNode, VisitState& state ) final;
-    DashNodeVector& nodeVector_;
-};
+                             NodeIds& nodeIds )
+        : RenderNodeVisitor( dashTree ),
+          _nodeIds( nodeIds )
+    {}
 
-VisibleCollectorVisitor::VisibleCollectorVisitor(
-        DashTreePtr dashTree,
-        DashNodeVector& nodeVector )
-    : RenderNodeVisitor( dashTree ),
-      nodeVector_( nodeVector )
-{ }
+    void visit( DashRenderNode& renderNode, VisitState& state ) final
+    {
+        const LODNode& lodNode = renderNode.getLODNode();
+        if( !lodNode.isValid( ))
+            return;
 
-void VisibleCollectorVisitor::visit( DashRenderNode& renderNode, VisitState& state )
-{
-    const LODNode& lodNode = renderNode.getLODNode();
-    if( !lodNode.isValid( ))
-        return;
+        if( !renderNode.isInFrustum( ))
+        {
+            state.setVisitChild( false );
+            return;
+        }
 
-    if( !renderNode.isInFrustum( ))
-    {
-        state.setVisitChild( false );
-        return;
+        if( renderNode.isVisible( ))
+        {
+            _nodeIds.push_back( lodNode.getNodeId( ));
+            state.setVisitChild( false );
+        }
```

Comment:
Why two variables: 
1) I did not wanted to do the same test twice.
2) If there is only one variable and is false along the tree, it will traverse to the smallest node.

As soon as I add the new node selection algorithm there will be no need for traversal. THose two variables etc